test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Query/Expressions/SelectExpandBinderTest.cs

```diff
@@ -304,6 +304,97 @@ public void ProjectAsWrapper_Element_ProjectedValueContainsModelID()
             Assert.Same(_model.Model, ModelContainer.GetModel(customerWrapper.ModelID));
         }
 
+        [Fact]
+        public void ProjectAsWrapper_Element_ProjectedValueContainsSubKeys_IfDollarRefInDollarExpand()
+        {
+            // Arrange
+            _model.Model.SetAnnotationValue<ClrTypeAnnotation>(_model.SpecialOrder, new ClrTypeAnnotation(typeof(SpecialOrder)));
+            IPropertyMapper propertyMapper = new IdentityPropertyMapper();
+
+            Customer customer = new Customer
+            {
+                Orders = new[]
+                {
+                    new Order { ID = 42 },
+                    new SpecialOrder { ID = 38 }
+                }
+            };
+            ODataQueryOptionParser parser = new ODataQueryOptionParser(
+                _model.Model,
+                _model.Customer,
+                _model.Customers,
+                new Dictionary<string, string> { { "$expand", "Orders/$ref" } });
```

Comment:
>"Orders/$ref" [](start = 62, length = 13)

please add tests for query options applied to /$ref. i.e.; Orders/$ref($filter=xyz), Orders/$ref($orderby=xyz), Orders/$ref($top=10), etc...

Reply:
Added. See test cases :
ProjectAsWrapper_Element_ProjectedValueContainsSubKeys_IfDollarRefInDollarExpand_AndNestedFilterClause

ProjectAsWrapper_Collection_ProjectedValueContainsSubKeys_IfDollarRefInDollarExpand_AndNestedTopAndSkip

---
In reply to: [298739015](https://github.com/OData/WebApi/pull/1797#discussion_r298739015) [](ancestors = 298739015)

Reply:
Great; thanks!

---
In reply to: [299181060](https://github.com/OData/WebApi/pull/1797#discussion_r299181060) [](ancestors = 299181060,298739015)